### ISSUE TITLE
Manifest.json should indicate if the asset should load or not.

### ISSUE THREAD

**Issue by najadojo** [KhronosGroup/glTF-Asset-Generator#429]
In particular Compatibility_05.gltf shouldn't load; the json should have a field saying so.

**Comment by bghgary** [comment]
We've talked about doing this, but haven't nailed down on what the right schema is yet. We eventually will want a set of negative tests that may or may not load in an implementation. TBD how that will be handled.

**Comment by stevk** [comment]
False closure. We've added a flag to show if models are valid glTF models, but Compatibility_05.gltf is a valid model even though it won't load.

**Comment by vpenades** [comment]
I also need this feature, since I am beginning to use the generated assets for automatic unit testing.

I understand there's a distinction between a gltf file being valid, and being loadable. Obviously, an invalid gltf should not load, but a valid gltf should not neccesarily mean it can be loaded.

So, my suggestion would be to add two additional properties to the manifest:

"valid": true | false
"loadable": true | false

For example, a model with an unknown required extensions should be defined as `valid=true` and `loadable=false`

**Comment by stevk** [comment]
I'll start work on adding this next. 

I agree with your suggested implementation of a "loadable" flag. On the other hand a "valid" flag shouldn't be needed due to all invalid models being seperated out into the [Negative tests folder](../tree/master/Output/Negative).

As a side note, currently this project is only creating two models that are valid and won't load:
- Compatibility_04
- Compatibility_05

**Comment by vpenades** [comment]
@stevk I think the "valid" flag can be useful to tell whether the file failed to load because some malformed json code, or because it has a feature that prevents it from loading correctly.

I am using the downloaded assets located here: [GeneratedAssets-0.6.0.zip](https://github.com/KhronosGroup/glTF-Asset-Generator/releases/download/v0.6.0/GeneratedAssets-0.6.0.zip) , not the actual output of the repository... as far as I can tell, the negative test files are not included.

Also, in the future it is expected more tests will be included, which might include malformed or incomplete JSON documents, where the "valid" flag should be false, so a unit test can ensure that the parser catched it correctly.

In other words: my current implementation throws two kinds of exceptions on load: JSON Parsing exception, and Unsupported extension exception, potentially there can be more cases, but it could be very useful to tell the unit tests which exception to expect.

**Comment by bghgary** [comment]
> I am using the downloaded assets located here: GeneratedAssets-0.6.0.zip , not the actual output of the repository... as far as I can tell, the negative test files are not included.

We are still working this, so it won't be in a released version yet. If you want the zipped version, we can release a new version when things are ready.

> in the future it is expected more tests will be included, which might include malformed or incomplete JSON documents, where the "valid" flag should be false, so a unit test can ensure that the parser catched it correctly.

All invalid assets will be put in a separate folder with a separate manifest. If we add a "valid" flag, the positive tests folder will all have this flag set to true and the negative tests folder will all have this flag set to false. I don't think this flag will be useful in that context.

All the tests in the positive tests folder should succeed in an implementation except for the ones marked with loadable as false.

**Comment by vpenades** [comment]
@bghgary 

> We are still working this, so it won't be in a released version yet. If you want the zipped version, we can release a new version when things are ready.

Yes please, I am trying not to hammer too much the repository itself

**Comment by bghgary** [comment]
Sure, we can do that. If the reasoning is not to hammer the repository, you can also copy the files from the Output folder directly.

**Comment by vpenades** [comment]
@bghgary My unit tests clone several repositories when run, but it is a hack of sorts... having a zip available is much more convenient, and it's supposed to be the curated data for testing.

**Comment by bghgary** [comment]
> having a zip available is much more convenient

Yup, not arguing that. I'm just saying you can also easily get the latest if you don't want to wait. We will release a version as soon as we are ready.

**Comment by stevk** [comment]
@vpenades We've released a version that we think will fit your needs. 

https://github.com/KhronosGroup/glTF-Asset-Generator/releases/tag/v0.6.1
